lnwallet/channel.go

```diff
@@ -6055,20 +6055,30 @@ func (lc *LightningChannel) availableCommitmentBalance(view *htlcView) (
 		ourBalance = 0
 	}
 
-	// Given the commitment weight, find the commitment fee in case of no
-	// added HTLC output.
+	// Calculate the commitment fee in the case where we would add another
+	// HTLC to the commitment, as only the balance remaining after this fee
+	// has been paid is actually available for sending.
 	feePerKw := filteredView.feePerKw
-	baseCommitFee := lnwire.NewMSatFromSatoshis(
-		feePerKw.FeeForWeight(commitWeight),
+	htlcCommitFee := lnwire.NewMSatFromSatoshis(
```

Comment:
Similar to the `predictAdded` field in `validateCommitmentSanity`, shouldn't we only conditionally add this weight if we're attempting to look ahead by one HTLC to see the expected balance _after_ this HTLC is added.

Reply:
Or is the assumption that this method is only used atm by the switch when it _is_ attempting to add a new HTLC to the commitment transaction?

Reply:
Non blocking, more of a side point, but if we _do_ add this param, then we can use this to display the current balance of a channel in `ListChannels` and `WalletBalance`, if we opt to just mask the concept of a channel reserver all together. If we go this route, then at least the user's wallet will show zero satoshis, rather than showing a balance which can't actually be sent/received (though there're other implications).

Reply:
> Or is the assumption that this method is only used atm by the switch when it _is_ attempting to add a new HTLC to the commitment transaction?

Yeah, that is the assumption, that it returns the "balance available for HTLCs". And that has to include the HTLC fee if we are the initiator.